File: web-frontend/modules/dashboard/searchTypes.js
import { BaseSearchType } from '@baserow/modules/core/search/types/base'

export class DashboardSearchType extends BaseSearchType {
  constructor(context = {}) {
    super(context)
    this.type = 'dashboard'
    this.name = 'Dashboard'
    this.icon = 'baserow-icon-dashboard'
    this.priority = 3
  }

  _getApplicationId(result) {
    const id = parseInt(result?.metadata?.application_id || result?.id)
    return isNaN(id) ? null : id
  }

  buildUrl(result, context = null) {
    const appId = this._getApplicationId(result)
    if (!appId) {
      return null
    }
    return {
      name: 'dashboard-application',
      params: { dashboardId: appId },
    }
  }

  isNavigable(result, context = null) {
    return this._getApplicationId(result) !== null
  }
}

File: backend/src/baserow/contrib/dashboard/search_types.py
from baserow.contrib.dashboard.models import Dashboard
from baserow.core.search.search_types import ApplicationSearchType


class DashboardSearchType(ApplicationSearchType):
    """
    Searchable item type specifically for dashboards.
    """

    type = "dashboard"
    name = "Dashboard"
    model_class = Dashboard
    priority = 3
